mozetl/topline/historical_backfill.py

```diff
@@ -57,16 +56,42 @@ def backfill_topline_summary(historical_df, path):
     df = df.select(*[F.col(f.name).cast(f.dataType).alias(f.name)
                      for f in topline_schema.fields])
 
+    logging.info("Saving historical data to {}.".format(path))
+
+    write_mode = "overwrite" if overwrite else "error"
+    writer = df.write.mode(write_mode)
+
     # Use the same parititoning scheme as topline_summary
-    df.write.partitionBy('report_start').parquet(path)
+    if batch:
+        writer = writer.partitionBy('report_start')
+    else:
+        # The csv file should be read from
+        # telemetry-private-analysis=2/executive-report. This folder contains csv
+        # files that should contain only a single date. Assert that property here.
+        path = "{}/report_start={}".format(path, df.head().report_start)
+        if df.select('report_start').distinct().count() != 1:
+            raise RuntimeError("There should only be a single reporting date")
+
+    writer.parquet(path)
 
 
 @click.command()
 @click.argument('source_s3_path')
 @click.argument('mode', type=click.Choice(['weekly', 'monthly']))
 @click.argument('bucket')
 @click.option('--prefix', default='topline_summary/v1')
-def main(source_s3_path, mode, bucket, prefix):
+@click.option('--batch/--no-batch', default=False,
+        help='Used when the csv file contains multiple reporting periods')
+@click.option('--overwrite/--no-overwrite', default=False,
+        help='Overwrite existing data. Caution when applying this batch mode,'
+             'since it will overwrite all data in the repository.')
```

Comment:
The word `repository` here might be confusing. Maybe use something like `in the specified S3 path`.